src/mlpack/methods/sparse_autoencoder/maximal_inputs_impl.hpp

```diff
@@ -20,8 +20,8 @@ inline void MaximalInputs(const arma::mat& parameters, arma::mat& output)
 {
   arma::mat paramTemp(parameters.submat(0, 0, (parameters.n_rows - 1) / 2 - 1,
                                         parameters.n_cols - 2).t());
-  double const mean = arma::mean(arma::mean(paramTemp));
-  paramTemp -= mean;
+  double const mean_ = arma::mean(arma::mean(paramTemp));
+  paramTemp -= mean_;
```

Comment:
```suggestion
  paramTemp -= arma::mean(arma::mean(paramTemp));
```
Why not just remove the temporary variable entirely?